### PR TITLE
Improve documentation for DIDComm mediator

### PR DESCRIPTION
**Pull request by Tekum-Emmanuella**
- Enhanced README.md with detailed setup and usage instructions
- Added examples and troubleshooting tips

Please review the changes and provide feedback.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,3 +55,76 @@ You can install them using the following command:
...
+ * Cargo
+
+## Setup
+1. Create a working directory eg(cd didcomm-mediator-rs) and cd into your directory.
+
+ 2. Clone the repository using the following command:
+
+git clone https://github.com/adorsys/didcomm-mediator-rs.git
+
+## Configuration Options
```
Comment (Christiantyemele) [inline README.md:73]: I dont't think this is neccessary

File: README.md [hunk 2]
```diff
@@ -55,3 +55,76 @@ You can install them using the following command:
...
+If you encounter any issues while running the application, here are some troubleshooting tips to help you resolve them
+ 
+ ### Common Issues
+
+  1. Build Errors:
+  * Ensure that you have the required system packages installed. `libssl-dev` and `pkg-config` 
+  * Ensure that you have the latest version of Rust and Cargo installed.
+  * CHeck for any missing dependencies using `cargo check`.
+
+  2. Runtime Errors:
```
Comment (Christiantyemele) [inline README.md:90]: you should focus on the setup and installation of the project and not the runtime issues

File: README.md [hunk 3]
```diff
@@ -55,3 +55,76 @@ You can install them using the following command:
...
+  * Ensure that the configuration file is valid and that all required fields are present.
+  * Ensure the mediator service is running and accessible.
+
+### Solutons
+ Check the missing dependencies 
+
+### Configuration Errors:
```
Comment (Christiantyemele) [inline README.md:97]: I dont't understand who is checking for typo in what

File: README.md [hunk 4]
```diff
@@ -55,3 +55,76 @@ You can install them using the following command:
...
+ * Double-check configuration files for typos or missing entries.
+ * Refer to the crate-specific documentation for required configuration settings.
+
+## Examples
+### Sending a DIDComm Message
```
Comment (Christiantyemele) [inline README.md:102]: This is good but should be in the general guide of the setup

**Review by Christiantyemele — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -55,3 +55,76 @@ You can install them using the following command:
...
 sudo apt install -y libssl-dev pkg-config
 ```
+
+## Step-by-Step Guides
+
+### Prerequisites
+Ensure you have the following installed:
+ * Rust
+ * Cargo
+
```
Comment (Christiantyemele) [inline README.md:65]: you should better give a link on how to install it

File: README.md [hunk 6]
```diff
@@ -55,3 +55,76 @@ You can install them using the following command:
...
+ 2. Clone the repository using the following command:
+
+git clone https://github.com/adorsys/didcomm-mediator-rs.git
+
+## Configuration Options
+The following configuration options are available:
+* `--config-file`: Path to the configuration file. Defaults to `config.toml`.
+* `--config`: Configuration options in TOML format. Overrides the configuration file.
+* `--log-level`: Log level. Can be one of `trace`, `debug`, `
+info`, `warn`, `error`, or `off`. Defaults to `info`.
```
Comment (Christiantyemele) [inline README.md:78]: I dont't think this is neccessary

File: README.md (hunk 3 above)
Comment (Christiantyemele) [inline README.md:97]: you should focus on the setup and installation of the project and not the runtime issues

File: README.md [hunk 7]
```diff
@@ -55,3 +55,76 @@ You can install them using the following command:
...
+-d '{
+    "id": "1234567890",
+    "type": "https://didcomm.org/routing/2.0/forward",
+    "to": ["did:example:bob"],
+    "message": {
+        "@id": "abcdefg",
+        "@type": "https://didcomm.org/routing/2.0/forward",
+        "to": ["did:example:alice"],
+        "message": {
+            "@id": "123",
```
Comment (Christiantyemele) [inline README.md:121]: This is good but should be in the general guide of the setup

**Review by Christiantyemele — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (Blindspot22) [reply README.md:78]: @Christiantyemele Should it be removed or what?

File: README.md [hunk 8]
```diff
@@ -127,4 +127,4 @@ curl -X POST http://localhost:8000/messages \
 This example demonstrates sending a routing message to the mediator, which it will forward to the intended recipient.
```
Comment (Christiantyemele) [inline README.md:127]: Where is the example

**Review by Christiantyemele — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (Christiantyemele) [reply README.md:77]: Yes i think it should be removed

File: README.md (hunk 3 above)
Comment (Blindspot22) [reply README.md:89]: Meaning the Runtime Errors, Configuration Errors are unnecessary too?

File: README.md (hunk 7 above)
Comment (Blindspot22) [reply README.md:102]: Sure I agree @Christiantyemele . But any proposed file where it could be?

Comment (Blindspot22) [reply README.md:102]: I think the 1. Start the mediator service:  
should stay but the 
2. Send a message using curl is the one to be displaced or removed

File: README.md (hunk 8 above)
Comment (Blindspot22) [reply README.md]: It's not necessary since you asked for it to be removed and placed in another file.

**Review by Christiantyemele — APPROVED** [review]
LGTM

**Review by Blindspot22 — APPROVED** [review]
👍

**Pull request merged**